src/structure.js

```diff
@@ -267,8 +264,7 @@ Structure.prototype.reference = function (path) {
      * @returns {Void}
      */
     unobserveAll: function () {
-      removeAllListenersBut(self, pathId, referenceListeners, changeListener);
-      referenceListeners = [changeListener];
+      removeAllListenersBut(listenerNs, changeListener);
```

Comment:
IIRC, the original API for `unobserveAll` only removed observers added through a cursor reference. This removes listeners added by other cursor references of the same keypath.